app/services/onlinetesting/OnlineTestExpiryService.scala

```diff
@@ -46,29 +46,67 @@ class OnlineTestExpiryServiceImpl(
 
   private implicit def headerCarrier = newHeaderCarrier
 
+  override def processNextTestForReminder(reminder: ReminderNotice): Future[Unit] = {
+    otRepository.nextTestForReminder(reminder).flatMap {
+      case Some(expiringTest) => processReminder(expiringTest, reminder)
+      case None => Future.successful(())
+    }
+  }
+
   override def processNextExpiredTest(): Future[Unit] = {
     otRepository.nextExpiringApplication.flatMap {
       case Some(expiredTest) => processExpiredTest(expiredTest)
       case None => Future.successful(())
     }
   }
 
-  override def processExpiredTest(expiringTest: ExpiringOnlineTest): Future[Unit] =
+  override def commitExpiredProgressStatus(expiringTest: ExpiringOnlineTest): Future[Unit] =
+    applicationRepository.addProgressStatusAndUpdateAppStatus(expiringTest.applicationId, PHASE1_TESTS_EXPIRED).map { _ =>
+      audit("ExpiredOnlineTest", expiringTest)
+    }
+
+  private def processReminder(expiringTest: NotificationExpiringOnlineTest, reminder: ReminderNotice): Future[Unit] =
+    for {
+      emailAddress <- candidateEmailAddress(expiringTest.userId)
+      _ <- emailCandidateForExpiringTestReminder(expiringTest, emailAddress, reminder)
```

Comment:
I think there was a small conversation last week about always emailing after database activity.

With this ordering if for some reason the database write files we'll end up spamming the user (as the scheduler will re-run, send the email, fail to write to the DB again).

If the email is last the failure to write to the DB will stop the email sending and the scheduler will eventually recover